src/sql/sql-query/sql-query.ts

```diff
@@ -0,0 +1,301 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Separator, SqlAliasRef, SqlLiteral, SqlRef } from '../index';
+import { SqlBase, SqlBaseValue } from '../sql-base';
+
+export interface SqlQueryValue extends SqlBaseValue {
```

Comment:
At this point I'd be tempted to introduce some expression types and nest them here. e.g., `SqlQueryValue` could have a `withExpression` which is an object with a `withKeyword`, `withUnits`, and `withSeparators`.